src/components/ModalExport.js

```diff
@@ -0,0 +1,166 @@
+import React, { useState, useEffect } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import { Modal } from './Modal.js'
+import { DropDownMenu } from './DropDownMenu.js'
+import ArrowDownWhite from '../images/keyboard_arrow_down_352466.svg'
+import ArrowDownBlack from '../images/iconfinder_ic_keyboard_arrow_down_black_352466.svg'
+//  util's.js
+import {
+  ellipsize,
+  headValue,
+  getDescendants,
+  download,
+  exportContext,
+  pathToContext,
+  timestamp
+} from '../util'
+
+const formatOptions = [{
+  id: 1,
+  type: 'Plain Text'
+}, {
+  id: 2,
+  type: 'HTML'
+}]
+
+export const ModalExport = () => {
+  const dispatch = useDispatch()
+  const cursor = useSelector(state => state.cursor)
+  const settings = useSelector(state => state.settings)
+  const [format, setFormat] = useState({ f1: 'plaintext', f2: 'Plain Text' })
+  const [isOpen, handleMenu] = useState(false)
+  const [wrapperRef, setWrapper] = useState()
+  const imgToShow = settings.dark ? ArrowDownWhite : ArrowDownBlack
+  const subsOrSub = getDescendants(cursor).length === 1 ? 'subthought' : 'subthoughts'
+  const exportInfo = `"${ellipsize(headValue(cursor))}" and ${getDescendants(cursor).length} ${subsOrSub} as ${format.f2} `
```

Comment:
You will want to call `getDescendants(cursor)` once and store the result for performance reasons, rather than calling it multiple times.